libs/cart/state/src/effects/cart-coupon.effects.spec.ts

```diff
@@ -132,6 +137,21 @@ describe('Daffodil | Cart | CartCouponEffects', () => {
       it('should dispatch a CartCouponApplyFailure action', () => {
         expect(effects.apply$).toBeObservable(expected);
       });
+
+      describe('because of an invalid coupon', () => {
+        beforeEach(() => {
+          const error = new DaffInvalidCouponCodeError('Invalid coupon code', mockCoupon.code);
```

Comment:
I may be wrong, but I think you need to remove the `mockCoupon.code` from this error in order to test the code you added. Because the new code is what adds the coupon code to the error, right? When the error comes in, it doesn't have the coupon code; otherwise you wouldn't need to set it in the effect.